examples/nlp/language_modeling/megatron_gpt_eval.py

```diff
@@ -213,6 +213,8 @@ def main(cfg) -> None:
             pretrained_cfg.precision = trainer.precision
             if trainer.precision == "16":
                 pretrained_cfg.megatron_amp_O2 = False
+            elif trainer.precision in ['bf16', 'bf16-mixed'] and cfg.get('megatron_amp_O2', False):
```

Comment:
I think we can enable this for fp16 as well. So something like:
```
if trainer.precision in [16 '16', '16-mixed', 'bf16', 'bf16-mixed'] and cfg.get('megatron_amp_O2', False):
    pretrained_cfg.megatron_amp_O2 = True
else:
    pretrained_cfg.megatron_amp_O2 = False
```

Reply:
https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/models/language_modeling/megatron_base_model.py#L767
fp16 not supported in amp O2.